transactionoutbox-core/src/main/java/com/synaos/transactionoutbox/TransactionOutboxImpl.java

```diff
@@ -156,41 +153,46 @@ private void expireIdempotencyProtection(Instant now) {
     int recordsDeleted;
     do {
       recordsDeleted =
-          transactionManager.inTransactionReturns(
-              tx ->
-                  uncheckedly(() -> persistor.deleteProcessedAndExpired(tx, flushBatchSize, now)));
+              transactionManager.inTransactionReturns(
+                      tx ->
+                              Utils.uncheckedly(() -> persistor.deleteProcessedAndExpired(tx, flushBatchSize, now)));
       totalRecordsDeleted += recordsDeleted;
     } while (recordsDeleted > 0);
     if (totalRecordsDeleted > 0) {
       String duration =
-          String.format(
-              "%dd:%02dh:%02dm:%02ds",
-              retentionThreshold.toDaysPart(),
-              retentionThreshold.toHoursPart(),
-              retentionThreshold.toMinutesPart(),
-              retentionThreshold.toSecondsPart());
+              String.format(
+                      "%dd:%02dh:%02dm:%02ds",
+                      retentionThreshold.toDaysPart(),
+                      retentionThreshold.toHoursPart(),
+                      retentionThreshold.toMinutesPart(),
+                      retentionThreshold.toSecondsPart());
       log.info(
-          "Expired idempotency protection on {} requests completed more than {} ago",
-          totalRecordsDeleted,
-          duration);
+              "Expired idempotency protection on {} requests completed more than {} ago",
+              totalRecordsDeleted,
+              duration);
     } else {
       log.debug("No records found to delete as of {}", now);
     }
   }
 
+  @Override
```

Comment:
Two public schedule methods?

Reply:
We need to overload these so you are able to set a groupId / requestId if this is not wanted we use a default value